src/control/server/instance.go

```diff
@@ -273,16 +273,22 @@ func (ei *EngineInstance) handleReady(ctx context.Context, ready *srvpb.NotifyRe
 	return ei.SetupRank(ctx, r, mapVersion)
 }
 
-func (ei *EngineInstance) SetupRank(ctx context.Context, rank ranklist.Rank, map_version uint32) error {
-	if err := ei.callSetRank(ctx, rank, map_version); err != nil {
+func (ei *EngineInstance) SetupRank(ctx context.Context, rank ranklist.Rank, map_version uint32) (err error) {
+	ei.ready.SetTrue()
```

Comment:
Doesn't this potentially create a gap where ready is set to true, but the setup hasn't happened yet?

Reply:
I share this concern. @tanabarr, since this PR changes the meaning of "ready" from state X to state Y (as defined by my previous PR comment), I wonder if we could skip the "ready" check in EngineInstance.CallDrpc for `SetRank` and `SetUp` dRPCs? E.g.,

    if method != SetRank && method != SetUp && !ei.IsReady() {
        return nil, errEngineNotReady
    }

Then we would no longer need the change here in `EngineInstance.SetupRank`. Is that right?